olmo/data/collator.py

```diff
@@ -138,3 +138,37 @@ def __call__(self, items: Union[List[Dict[str, Any]], List[torch.Tensor]]) -> Di
             out["metadata"] = all_metadata
 
         return out
+
+
+@dataclass
+class CustomDatasetDataCollator(DataCollator):
+    input_id_field: str = "input_ids"
+    attention_mask_field: Optional[str] = None
+    attention_bias_field: Optional[str] = None
+    label_mask_field: Optional[str] = None
+    index_field: Optional[str] = None
+    instance_mask_field: Optional[str] = None
+    doc_lens_field: Optional[str] = None
+    metadata_field: Optional[str] = None
+
+    def _relabel_fields(self, items: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+        return [self._relabel_item(x) for x in items]
+
+    def _relabel_item(self, item: Dict[str, Any]) -> Dict[str, Any]:
+        out = {
+            "input_ids": item.__getitem__(self.input_id_field),
+            "attention_mask": item.__getitem__(self.attention_mask_field) if self.attention_mask_field else None,
+            "attention_bias": item.__getitem__(self.attention_bias_field) if self.attention_bias_field else None,
+            "label_mask": item.__getitem__(self.label_mask_field) if self.label_mask_field else None,
+            "index": item.__getitem__(self.index_field) if self.index_field else None,
+            "instance_mask": item.__getitem__(self.instance_mask_field) if self.instance_mask_field else None,
+            "metadata": item.__getitem__(self.metadata_field) if self.metadata_field else None,
```

Comment:
Why all these `item.__getitem__` calls? Why not just `item[]`?

Reply:
I can change this, was intially working trying to mirror the `.get` method used in the default collator, agree that `[]` is cleaner, can make the change

Reply:
Fixed